subprojects/execution/src/main/java/org/gradle/internal/execution/steps/IdentifyStep.java

```diff
@@ -51,42 +56,75 @@ public <T> Deferrable<Try<T>> executeDeferred(UnitOfWork work, C context, Cache<
 
     @Nonnull
     private IdentityContext createIdentityContext(UnitOfWork work, C context) {
-        InputFingerprinter.Result inputs = work.getInputFingerprinter().fingerprintInputProperties(
-            ImmutableSortedMap.of(),
-            ImmutableSortedMap.of(),
-            ImmutableSortedMap.of(),
-            ImmutableSortedMap.of(),
-            work::visitIdentityInputs
+        String workClassName = work.getClass().getName();
+        return operation(operationContext -> {
```

Comment:
Instead of changing this method (called `createIdentityContext`) to wrap the operation around internally, why not add a new method that calls this one inside the operation? That's a nice separation of concerns and we keep true to the name of the methods.

Reply:
(Also, if I get my way, I'll be able to remove the anonymous inner classes here, making this method unnecessary completely, but that's a separate story.)

Reply:
If by referring to the anonymous inner class you mean the one passed to the `setResult` method, then it does not seem to work due to Configuration Caching not being able to serialize lambdas. I tried doing that initially, but a significant number of integrations tests broke with CC executor.

Apart from suppressing directly, the only option of avoiding the pesky "convert to lambda" suggestion by the IDE is to actually create implementation classes. Would you like it better this way?

Reply:
I mean this about anonymous inner classes: https://github.com/gradle/gradle/pull/22219